packages/visual-editor/src/ui/app-templates/basic/index.ts

```diff
@@ -362,14 +392,19 @@ export class Template extends SignalWatcher(LitElement) implements AppTemplate {
       | symbol = nothing;
     const last = this.run.app.last?.last;
     if (last) {
-      const htmlOutput = isHTMLOutput(last);
+      const htmlOutput = getHTMLOutput(last);
       if (htmlOutput !== null) {
+        // Note there is another iframe within this one, which is what actually
+        // serves the generated html. This middle layer exists purely so that we
+        // can serve a different CSP for generated apps vs the main app. After
```

Comment:
Note the key ways that the CSP differs?

* Only has network access to CDN locations that the generation is prompted to use
* Disables TrustedTypes, but that's ok due to the limited network access, null origin, and sandbox